### PR TITLE
Galley item hide properties/offers

### LINKED ISSUE #4438: Hide Properties when they are empty
> Shouldn't we hide Properties when empty? (and show by default Description)

Yes, good point

We should hide Properties tab when there are no traits.

_Originally posted by @yangwao in https://github.com/kodadot/nft-gallery/issues/4360#issuecomment-1336191553_

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #4438
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
Hey! 
Let's start with something simple:
- disable `o-tab-item` when there is no `MAKE_OFFER` event in `nft.events`
- show tooltip for rmrk if prefix rmrk
- show tooltip for disabled when no event

File: components/rmrk/Gallery/Gallery.vue
Comment (daiagi) [reply components/rmrk/Gallery/Gallery.vue]: haha yes. it was a leftover. removed already

**Comment by daiagi** [comment]
> Hey! Let's start with something simple:
> 
>     * disable `o-tab-item` when there is no `MAKE_OFFER` event in `nft.events`
> 
>     * show tooltip for rmrk if prefix rmrk
> 
>     * show tooltip for disabled when no event

the tooltip is the heart of where it gets messy
in order to have tooltip on the tab header i need to provide o-teb-item with a header slot, agree?
because the header slot is a child of a disabled button generated by o-tab-item, it doesn't receive hover events
from there the road is set for all the css you see
maybe it's hackish, and i would love to have it simpler, but i see no alternative so far.

the real issue starts when extracting this into a component and needing to put o-tab-children where they belong

if you will permit it, i can push a working version where the required behavior is working, but it's not extracted into a component
but same as you, I don't like it, it should be in a component

should i uncomment the slot so you can see live what is happening?

**Comment by roiLeo** [comment]
> the tooltip is the heart of where it gets messy in order to have tooltip on the tab header i need to provide o-tab-item with a header slot, agree?

Yes, I would rather have directly change `GalleryItemTabsPanel` to include `<template #header>`

>  all the css you see maybe it's hackish, and i would love to have it simpler, but i see no alternative so far.

please, let's try to avoid as much as possible css hack 🥺 

> the real issue starts when extracting this into a component and needing to put o-tab-item's children where they belong

Are you refering to your `GalleryTab` component? IMO, it's not needed
Btw, oruga-ui shouldn't be used in kodadot app, it was originally planned for histoire components.

**Comment by daiagi** [comment]
Found it!
it happens when o-tab-item recieves no children
solution is to pass empty div as v-else

```vue
    <o-tab-item value="0" class="py-5">
      <template #header> Properties </template>
      <o-table
        v-if="nftMetadata?.attributes.length"
        :data="nftMetadata?.attributes"
        hoverable>
        <o-table-column v-slot="props" field="value" label="Trait">
          {{ props.row.value }}
        </o-table-column>
        <o-table-column
          v-slot="props"
          field="trait_type"
          :label="$t('tabs.tabProperties.section')">
          {{ props.row.trait_type }}
        </o-table-column>
      </o-table>
      <div v-else></div>
    </o-tab-item>
```

**Comment by roiLeo** [comment]
Sorry, I didn't knew tooltip wasn't working with disabled tab

Let me know if you find something, might be good to open an issue in oruga repo?

**Comment by daiagi** [comment]
> Sorry, I didn't knew tooltip wasn't working with disabled tab
> 
> Let me know if you find something, might be good to open an issue in oruga repo?

it's ok. I got it already
i'll submit working PR tommorow and we'll continue improving from that
Thank you :smile:

File: libs/ui/src/components/TooltipTab/TooltipTab.vue
Comment (roiLeo) [inline libs/ui/src/components/TooltipTab/TooltipTab.vue:8]: could you please use [oruga-ui tooltip](https://oruga.io/components/tooltip.html) to not mix framework

Comment (roiLeo) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue:8]: can't we seperate this two component to have something more atomic?

Comment (daiagi) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue:8]: you mean a component who is just the tooltip with some styles applies?

Comment (roiLeo) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue:8]: on one side the `o-tab-item` wrapper with template slot and on the other side the `o-tooltip` wrapper.
Since we'll need to reuse tooltip component.

**Comment by daiagi** [comment]
@roiLeo 
this assignment is proving much more challenging then it appears on the surface... :tired_face: 
I've switch to use oruga-ui tool tip and it working nicely in histoire
/home/luke/Videos/Screencasts/Screencast from 09-12-22 17:10:55.webm

but now not working in the app
by not working i mean not showing on hover

some more details that may help you help me...

if i ```always``` props  like this  in the histoire comonent
```vue
      <o-tooltip
        v-if="disabled"
        append-to-body
        always
        class="disabled-tab-tooltip"
        position="top"
        :label="disabledTooltip"
        @click.native.stop>
        {{ label }}
      </o-tooltip>
```

then when the page renders it looks

image


and then, when hovering over offers tab the tooltip "finds" it's position

image

any ideas?

File: libs/ui/src/components/TooltipTab/TooltipTab.vue
Comment (daiagi) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue]: ok. ill seperate them
but first need to figure out why toolyip doesn't show on hover in the app but does show in histoire

Comment (roiLeo) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue]: > ok. ill seperate them but first need to figure out why tooltip doesn't show on hover in the app but does show in histoire

might be some css issue on oruga import? I had a similar problem where I didn't have the same rendering with the app

Comment (daiagi) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue]: i'm playing around with that but with no result /conclusion
can you provide more direction please?

Comment (daiagi) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue]: ok i'm closing in on it
the issue comes from somewhere in the themes scss files

Comment (roiLeo) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue]: I am also trying to find a solution, I can see that it works on histoire.
We can try to inspect histoire element & GalleryItem tab to compare the difference.

Comment (daiagi) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue]: found it

```scss
  body,
  div#__nuxt,
  div#__layout,
  main {
    margin: 0;
    width: 100%;
    height: 100%;
    background-color: $background-dark;
// this line here
    // z-index: 1;
  }
```

mmm should i just delete it? it must have been put there for a reason

Comment (roiLeo) [reply libs/ui/src/components/TooltipTab/TooltipTab.vue]: oh my... nice find! yeah I would remove it it has been introduced in https://github.com/kodadot/nft-gallery/pull/4440 and I think has nothing to do there

**Comment by daiagi** [comment]
@roiLeo 
I think we're good to go
last thing i think is that the tooltip story doesn't really work :cry: 
I found it hard to make the tooltip show there

can we mark is as leftovers?

**Comment by exezbcz** [comment]
Looks good so far 
image
However, the inactive tab should not have a grey stroke, only the text color grey.

- typeface of the tooltip doesnt look like its worksans

File: libs/ui/src/components/NeoTooltip/NeoTooltip.scss
Comment (roiLeo) [inline libs/ui/src/components/NeoTooltip/NeoTooltip.scss:15]:
```suggestion
        font-family: 'Work Sans';
```

Comment (roiLeo) [reply libs/ui/src/components/NeoTooltip/NeoTooltip.scss:15]: Capture d’écran 2022-12-12 à 8 42 50 AM

**Review by roiLeo — COMMENTED** [review]
Last thing, you'll need to fix typo and it's good to go.
I think we can increase the bounty for the work done here.

~~FYI, it's possible to override some [o-tip variable](https://oruga.io/components/tooltip.html#style)~~ nvm...

**Comment by daiagi** [comment]
> Last thing, you'll need to fix typo and it's good to go. I think we can increase the bounty for the work done here.

Done. Thank you :+1:

**Review by roiLeo — APPROVED** [review]
Nice work!

✅ code lgtm

**Comment by roiLeo** [comment]
> The message "Offers are not available on rmrk" does not appear when I hover over the offers on this nft

Not related with this issue, might be coming from #4463 


Capture d’écran 2022-12-12 à 2 08 17 PM

**Comment by exezbcz** [comment]
> > The message "Offers are not available on rmrk" does not appear when I hover over the offers on this nft
> 
> Not related with this issue, might be coming from #4463
> 
> Capture d’écran 2022-12-12 à 2 08 17 PM

good to know, thanks for the clarification!

**Comment by daiagi** [comment]
Ready to go isn't it?

**Comment by roiLeo** [comment]
> Ready to go isn't it?

You might take a look at https://github.com/kodadot/nft-gallery/issues/4480#issuecomment-1346864543

**Comment by daiagi** [comment]
> > Ready to go isn't it?
> 
> You might take a look at [#4480 (comment)](https://github.com/kodadot/nft-gallery/issues/4480#issuecomment-1346864543)


I see... anything i can do to help resolve that?

**Comment by roiLeo** [comment]
> > > Ready to go isn't it?
> > 
> > 
> > You might take a look at [#4480 (comment)](https://github.com/kodadot/nft-gallery/issues/4480#issuecomment-1346864543)
> 
> I see... anything i can do to help resolve that?

Could you test when setting `z-index: 100;` on main if it's working as excepted and if there is no side effects?

**Comment by roiLeo** [comment]
> @roiLeo the only side effect I've found so far is this:

- can be fixed with a background-color

> 
> on a related issue... how do we deal with responsive font-size (to prevent tooltip from overflowing the screen on mobile) I expermineted and found this to work
> 
> ```css-scss
>     .o-tip__content {
>         font-size: max(min(2vw, 0.9rem), 0.45rem);
> ```
> 
> but maybe this is not the preferred method?

It won't really be displayed on mobile, but yes it seems to me to be a good solution as long as you don't use px

**Comment by daiagi** [comment]
> > @roiLeo the only side effect I've found so far is this:
> 
>     * can be fixed with a background-color


mmm it actually has background already

image

**Comment by daiagi** [comment]
> works for me:

with main z-index:100 ?

**Comment by daiagi** [comment]
@roiLeo is there anything more i need to do here?
do you want me to set z-index:100 as part of this PR?

**Comment by roiLeo** [comment]
> @roiLeo is there anything more i need to do here? do you want me to set z-index:100 as part of this PR?

Would be top if you could fix https://deploy-preview-4460--koda-nuxt.netlify.app/rmrk/gallery/15527227-B02B8226FEE3EAA06D-ART-BRIDGE-0000000000000001?redesign=true hover not visible.

Next, wait for a 2nd approval and it's good to go.

**Comment by daiagi** [comment]
> Would be top if you could fix https://deploy-preview-4460--koda-nuxt.netlify.app/rmrk/gallery/15527227-B02B8226FEE3EAA06D-ART-BRIDGE-0000000000000001?redesign=true hover not visible.

Done :heavy_check_mark:

**Review by prachi00 — APPROVED** [review]

**Comment by yangwao** [comment]
pay 60 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $60 @ 26.64 USD/KSM ~ 2.252 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x8c938dfb0435482e592b2b853515271702f2ecb0137d265184abfafe01620d30](https://kusama.subscan.io/extrinsic/0x8c938dfb0435482e592b2b853515271702f2ecb0137d265184abfafe01620d30)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by daiagi** [comment]
> let's resolve conflicts @daiagi and merge it! :)

Done

**Pull request merged**
